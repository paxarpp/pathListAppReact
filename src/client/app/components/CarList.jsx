import React, { Component } from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

import Car from './Car.jsx';

export default class CarList extends Component {
  render() {
    const { cars, selectedCar, deleteCarHandler } = this.props;
    return (
      <div className="carList">
        <ReactCSSTransitionGroup
          transitionName="anim"
          transitionAppear={false}
          transitionEnterTimeout={300}
          transitionLeaveTimeout={300}
          transitionEnter={true}
          transitionLeave={true}
        >
          {cars.map(car => {
            return (
              <div key={car.name} className="list">
                <Car
                  selectedCar={selectedCar}
                  car={car}
                  handler={this.props.carInfo(car.name)}
                  deleteCarHandler={deleteCarHandler}
                />
              </div>
            );
          })}
        </ReactCSSTransitionGroup>
      </div>
    );
  }
}
