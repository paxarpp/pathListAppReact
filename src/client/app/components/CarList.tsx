import React from 'react';
import styled from 'styled-components';
import Car from './Car';
import CountErr from '../components/CountErr';
import { IError, ICar } from './interfaces';

interface IProps {
  cars: ICar[];
  selectedCar: string;
  deleteCarHandler: () => void;
  carInfo: (name: string) => any;
  error: IError[][];
}

const prepareCount = (error: IError[][], car: ICar) => {
  const arr = error
    .filter((el)=> el.find(elem => elem.name === car.name) && el)
    .map(el => el.filter(path => (path.errorFuel || path.errorPath) && path));
  return arr[0] ? arr[0].length : 0;
};

const CarList = ({ cars, selectedCar, deleteCarHandler, error, carInfo }: IProps) => (
  <WrapperCarList>
    {cars.map(car => (
      <List key={car.name}>
        <Car selectedCar={selectedCar} car={car} handler={carInfo(car.name)} deleteCarHandler={deleteCarHandler} />
        <CountErr count={prepareCount(error, car)} />
      </List>
    ))}
  </WrapperCarList>
);

const WrapperCarList = styled.div`
  width: 300px;
  padding: 50px 0 10px 20px;
  background-color: #abf2eb7a;
  flex: 1;
`;

const List = styled.div`
  display: flex;
  cursor: pointer;
  width: 100%;
  height: 2.5rem;
  position: relative;
  :hover {
    text-shadow: 1px 1px 5px grey;
    transform: scale(1.01);
    transition: all 0.2s linear;
    cursor: pointer;
  }
`;

export default CarList;
