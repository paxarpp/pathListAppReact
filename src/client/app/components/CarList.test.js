import React from 'react';
import renderer from 'react-test-renderer';
import CarList from './CarList';

test('CarList has 2 car', () => {
  const cars = [{ name: 'test1' }, { name: 'test2' }];
  const error = [];
  const selectedCar = '';
  const component = renderer.create(
    <CarList deleteCarHandler={() => {}} carInfo={() => {}} selectedCar={selectedCar} error={error} cars={cars} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CarList has 2 car, select test2', () => {
  const cars = [{ name: 'test1' }, { name: 'test2' }];
  const error = [];
  const selectedCar = 'test2';
  const component = renderer.create(
    <CarList deleteCarHandler={() => {}} carInfo={() => {}} selectedCar={selectedCar} error={error} cars={cars} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CarList has 2 car, select test1, error1', () => {
  const cars = [{ name: 'test1' }, { name: 'test2' }];
  const error = [
    {
      name: 'test1',
      first: true,
      fuel: false,
      path: true
    },
    {
      name: 'test1',
      first: false,
      fuel: false,
      path: true
    }
  ];
  const selectedCar = 'test1';
  const component = renderer.create(
    <CarList deleteCarHandler={() => {}} carInfo={() => {}} selectedCar={selectedCar} error={error} cars={cars} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CarList has 2 car, no select, error1', () => {
  const cars = [{ name: 'test1' }, { name: 'test2' }];
  const error = [
    {
      name: 'test1',
      first: true,
      fuel: false,
      path: true
    },
    {
      name: 'test1',
      first: false,
      fuel: false,
      path: true
    }
  ];
  const selectedCar = '';
  const component = renderer.create(
    <CarList deleteCarHandler={() => {}} carInfo={() => {}} selectedCar={selectedCar} error={error} cars={cars} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
