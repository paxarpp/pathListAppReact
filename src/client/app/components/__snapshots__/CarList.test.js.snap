// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarList has 2 car 1`] = `
<div
  className="carList"
>
  <span>
    <div
      className="list"
    >
      <div
        className=""
      >
        test1
        <a
          className="delit"
        >
          ✖
        </a>
      </div>
    </div>
    <div
      className="list"
    >
      <div
        className=""
      >
        test2
        <a
          className="delit"
        >
          ✖
        </a>
      </div>
    </div>
  </span>
</div>
`;

exports[`CarList has 2 car, no select, error1 1`] = `
<div
  className="carList"
>
  <span>
    <div
      className="list"
    >
      <div
        className=""
      >
        test1
        <a
          className="delit"
        >
          ✖
        </a>
      </div>
      <span
        className="counterWrapError"
      >
        <span
          className="counter"
        >
          1
        </span>
      </span>
    </div>
    <div
      className="list"
    >
      <div
        className=""
      >
        test2
        <a
          className="delit"
        >
          ✖
        </a>
      </div>
    </div>
  </span>
</div>
`;

exports[`CarList has 2 car, select test1, error1 1`] = `
<div
  className="carList"
>
  <span>
    <div
      className="list"
    >
      <div
        className="selected"
      >
        test1
        <a
          className="delit"
        >
          ✖
        </a>
      </div>
      <span
        className="counterWrapError"
      >
        <span
          className="counter"
        >
          1
        </span>
      </span>
    </div>
    <div
      className="list"
    >
      <div
        className=""
      >
        test2
        <a
          className="delit"
        >
          ✖
        </a>
      </div>
    </div>
  </span>
</div>
`;

exports[`CarList has 2 car, select test2 1`] = `
<div
  className="carList"
>
  <span>
    <div
      className="list"
    >
      <div
        className=""
      >
        test1
        <a
          className="delit"
        >
          ✖
        </a>
      </div>
    </div>
    <div
      className="list"
    >
      <div
        className="selected"
      >
        test2
        <a
          className="delit"
        >
          ✖
        </a>
      </div>
    </div>
  </span>
</div>
`;
